### PR TITLE
Fix: add account should not permit duplicate email addresses

### LINKED ISSUE #2080: Add Account should not permit duplicate email address
If the Add Account dialog is given the same name and email address as an existing account it will create a second account duplicating the email address of the existing account as requested. (In testing this the encryption key created during the setup of the first account was selected for the new account - the creation of a new key was for the new account was not tested.) Then, in the home screen on the line for the new account, if the Trash icon is clicked, the program appears to start removing the duplicate account, but the display stops updating.

Having the same email address attached to two different accounts in the same pile does not make sense and is sure to cause problems. The Add Account dialog should screen the supplied email address and reject it if it is the same as the email address for an existing account. Preferably the email addresses should be canonicalized before comparison.

### PR DESCRIPTION
**Pull request by mnahinkhan**
This PR fixes #2080.

The changes made were:

1. AddProfile in .../mailpile/plugins/contacts.py now checks the new email address before adding a new profile. (We make sure it is not an edit, as suggested in the Issue). So if creating a new profile, the API call fails if the e-mail address already exists.

2. The index.html template collects the accounts passed to it in results.profiles to extract and store the email addresses in a global JS variable.

3. The template .../shared-data/default-theme/html/profiles/account-form.html uses that array to perform an additional validation check on the email entered (we created a new error logo color for this class of errors).

So in summary, both the front end and back end have been updated to fix this issue. Please find screenshots illustrating the fix:

Screen Shot 2020-12-07 at 5 05 51 PM

Screen Shot 2020-12-07 at 5 06 56 PM

### CODE REVIEW

**Comment by BjarniRunar** [comment]
Thank you so much for doing this, and please accept my apologies for ignoring this for so long.

I will probably merge this to my local tree and push up along with other changes once I have had time to test it.

**Comment by mnahinkhan** [comment]
Thank you! I would like to say that I had spotted a bug back then, which I thought to work on but didn't get the time to.

Unfortunately I have forgotten what the bug may have been. 

I know this is unhelpful but this is just to say I encourage you to test, as you stated you would 🙂.

In case you spot errors please feel free to state them here and I'll try to fix it (or if it's simple enough please feel free to fix of course). 

I'll also let you know in case it comes to my mind.
